src/main/java/org/folio/service/file/SplitFileProcessingService.java

```diff
@@ -227,17 +391,136 @@ public Future<Void> cancelJob(
+
+    client.postChangeManagerJobExecutions(request, promise::handle);
+
+    return promise
+      .future()
+      .map(this::verifyOkStatus)
+      .map(buffer ->
+        BufferMapper.mapBufferContentToEntitySync(
+          buffer,
+          InitJobExecutionsRsDto.class
+        )
+      );
+  }
+
+  /**
+   * Create parent job executions for all files described in the upload definition.
+   * @return a {@link Future} containing a map from filename/key -> {@link JobExecution}
+   */
+  protected Future<Map<String, JobExecution>> createParentJobExecutions(
+    ProcessFilesRqDto entity,
+    ChangeManagerClient client
+  ) {
+    InitJobExecutionsRqDto initJobExecutionsRqDto = new InitJobExecutionsRqDto()
+      .withFiles(
+        entity
+          .getUploadDefinition()
+          .getFileDefinitions()
+          .stream()
+          .map(FileDefinition::getSourcePath)
+          .map(key -> new File().withName(key))
+          .collect(Collectors.toList())
+      )
+      .withJobProfileInfo(entity.getJobProfileInfo())
+      .withSourceType(InitJobExecutionsRqDto.SourceType.COMPOSITE)
+      .withUserId(
+        getUserIdFromMetadata(entity.getUploadDefinition().getMetadata())
+      );
+
+    return sendJobExecutionRequest(client, initJobExecutionsRqDto)
+      .map((InitJobExecutionsRsDto response) -> {
+        LOGGER.info(
+          "Created {} parent job executions",
+          response.getJobExecutions().size()
+        );
+        // turn into map from filename -> JobExecution
+        return response
+          .getJobExecutions()
+          .stream()
+          .collect(Collectors.toMap(JobExecution::getSourcePath, exec -> exec));
+      })
+      .onFailure(err -> LOGGER.error("Error creating parent job execution", err)
+      );
+  }
+
+  /**
+   * Split a file into chunks, returning a {@link SplitFileInformation} object containing
+   * the original key, the total number of records in the file, and the keys of the split chunks
+   */
+  protected Future<SplitFileInformation> splitFile(String key) {
+    return Future
+      .succeededFuture(new SplitFileInformation().withKey(key))
+      // splitting and counting must be done sequentially as splitting deletes the original file
+      .compose(result ->
+        minioStorageService
+          .readFile(key)
+          .map((InputStream stream) -> {
+            try {
+              return result.withTotalRecords(
+                FileSplitUtilities.countRecordsInMarcFile(stream)
+              );
+            } catch (IOException e) {
+              throw new UncheckedIOException(e);
+            }
+          })
+      )
+      .compose(result ->
+        fileSplitService
+          .splitFileFromS3(vertx.getOrCreateContext(), key)
+          .map(result::withSplitKeys)
+      );
+  }
+
+  protected Buffer verifyOkStatus(HttpResponse<Buffer> response) {
+    if (response.statusCode() >= 200 && response.statusCode() <= 299) {
```

Comment:
nitpick: use the numerous enums available that represent HTTP statuses instead of ints